backend/src/test/java/com/github/redstylzz/backend/service/PaymentServiceTest.java

```diff
@@ -46,32 +54,38 @@ void shouldThrowExceptionIfCategoryDoesNotExist_OnAdd(){
 
     @Test
     void shouldReturnListAfterAddingPayment_OnAdd() {
-        Payment payment = TestDataProvider.testPayment();
-        String userID = payment.getUserID();
+        String randomUUID = UUID_STRING;
+        Payment actualPayment = TestDataProvider.testPayment();
+        actualPayment.setPaymentID(randomUUID);
+        String userID = actualPayment.getUserID();
         when(categoryRepository.existsByUserIDAndCategoryID(anyString(), anyString())).thenReturn(true);
-        when(paymentRepo.getAllByUserIDAndCategoryID(anyString(), anyString())).thenReturn(List.of(payment));
+        when(paymentRepo.getAllByUserIDAndCategoryID(anyString(), anyString())).thenReturn(List.of(actualPayment));
 
-        assertEquals(List.of(payment), underTest.addPayment(userID, payment));
+        assertEquals(List.of(Payment.convertPaymentToDTO(actualPayment)), underTest.addPayment(userID, actualPayment));
         verify(paymentRepo).save(any(Payment.class));
     }
 
     @Test
     void shouldThrowExceptionIfCategoryDoesNotExist_OnDelete() {
-        Payment payment = TestDataProvider.testPayment();
-        String userID = payment.getUserID();
+        Payment actualPayment = TestDataProvider.testPayment();
+        String userID = actualPayment.getUserID();
+        String categoryID = actualPayment.getCategoryID();
```

Comment:
if you need to reuse specific static values in your test class you should declare them as private static final attributes at the beginning of the test class instead of assigning them to a new variable in every single test 
static values in classes should also be in all upper case to indicate that they are static final variables, e.g.
private static final String USER_ID = "testUserId";